File: backend/src/main/kotlin/com/paymentschedule/model/PaymentScheduleResponse.kt
package com.paymentschedule.model

/**
 * Payment schedule response
 *
 * @property paymentScheduleLines
 * @property paymentScheduleTotals
 * @property purchaseOptionTotals
 * @property ibrNeeded
 * @constructor Create empty Payment schedule response
 */
data class PaymentScheduleResponse(
    val paymentScheduleLines: List<PaymentScheduleLine>,
    val purchaseOptionTotals: PurchaseOptionTotals,
    val paymentScheduleTotals: PaymentScheduleTotals,
    val ibrNeeded: Boolean,
)

File: frontend/src/types/payment-schedule/response/PaymentScheduleResponse.ts
import { PaymentScheduleLine } from './PaymentScheduleLine';
import { PaymentScheduleTotals } from './PaymentScheduleTotals';
import { PurchaseOptionTotals } from './PurchaseOptionsTotals';

export interface PaymentScheduleResponse {
  paymentScheduleLines: PaymentScheduleLine[];
  paymentScheduleTotals: PaymentScheduleTotals;
  purchaseOptionTotals: PurchaseOptionTotals;
  ibrNeeded: boolean;
}
